File: Python3/601 - 800/667 Beautiful Arrangement II.py
# 64ms, 77.67%; 14.1MB, 7.69%
class Solution:
    def constructArray(self, n: int, k: int) -> list:
        res = [0] * n
        res[0:k:2] = [x for x in range(1, 1 + (1 + k) // 2)]
        res[1:k:2] = [x for x in range(n, n - k // 2, -1)]
        delta = -1 if k > 1 and res[k - 2] < res[k - 1] else 1
        for i in range(k, n):
            res[i] = res[i - 1] + delta
        return res
    

File: 601 - 800/667. Beautiful Arrangement II.cc
// 24ms, 100.00%; 9.4MB, 100.00%
#include <bits/stdc++.h>
using namespace std;

static int x = []() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    return 0;
}();

class Solution {
public:
    vector<int> constructArray(int n, int k) {
        vector<int> res(n);
        for (int i = 0, x = 1, y = n; i < k; ++i) res[i] = (i & 1) ? y-- : x++;
        int delta = (k > 1 && res[k - 2] < res[k - 1]) ? -1 : 1;
        for (int i = k; i < n; ++i) res[i] = res[i - 1] + delta;
        return res;
    }
};
